pkg/cluster/cluster.go

```diff
@@ -118,12 +117,16 @@ func (c *Cluster) setup() error {
 		return fmt.Errorf("invalid cluster spec: %v", err)
 	}
 
-	var phase spec.ClusterPhase
+	// Recover the cached cluster state from the status in the current state.
+	// We could be using it to check if already "Failed" later.
```

Comment:
why dont we directly check c.cluster.State if we need a "correct view" for the restart case?